SharedLibrary/Sources/Shared/Grid.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "Shared/Types.hpp"
+#include "Shared/Array.hpp"
+#include <vector>
+#include <functional>
+
+namespace uf {
+
+struct GridTransformation {
+	vec3i shift = {0,0,0};
+	vec3i resize = {0,0,0};
+};
```

Comment:
There isn't obvious what arguments mean. I suggest to name these fields "deltaShift" and "deltaResize".